api/app/utils/geospatial.py

```diff
@@ -0,0 +1,165 @@
+    """
+    Raster data array and caller requested metadata
+    """
+
+    data_array: Any
+    data_geotransform: Optional[Any]
+    data_projection: Optional[Any]
+    data_x_size: Optional[int]
+    data_y_size: Optional[int]
+
+
+def get_geospatial_metadata(data_source, options: Optional[GeospatialOptions]):
+    """
+    Extracts metadata defined by the options parameter from the data_source raster
+
+    :param data_source: input raster to extract metadata from
+    :param options: defines the metadata the caller is interested in extracting from the raster
+    :return: optional metadata of geotransform, project, x and y sizes from the raster
+    """
+    geotransform = None
+    projection = None
+    x_size = None
+    y_size = None
+    if options is None:
+        return (geotransform, projection.x_size, y_size)
+
+    if options.include_geotransform:
+        geotransform = data_source.GetGeoTransform()
+
+    if options.include_projection:
+        projection = data_source.GetProjection()
+
+    if options.include_x_size:
+        x_size = data_source.RasterXSize
+
+    if options.include_y_size:
+        y_size = data_source.RasterYSize
+
+    return (geotransform, projection, x_size, y_size)
+
+
+def read_raster_data(data_source):
+    """
+    Read raster data and return as array from the data source
+
+    :param data_source: data source to read the data from
+    :return: raster data as array
+    """
+    data_band = data_source.GetRasterBand(1)
+    data_array = data_band.ReadAsArray()
+    return data_array
+
+
+def get_raster_data(data_source, options: Optional[GeospatialOptions]) -> GeospatialReadResult:
+    """
+    Returns the raster data and optional metadata the caller is interested defined by the options parameter
+
+    :param data_source: raster data source
+    :param options: options defining the metadata the caller is interested in
+    :return: raster data array and potential metadata
+    """
+    (data_geotransform, data_projection, data_x_size, data_y_size) = get_geospatial_metadata(data_source, options)
+    data_array = read_raster_data(data_source)
+    data_source = None
+    del data_source
+    return GeospatialReadResult(data_array=data_array, data_geotransform=data_geotransform, data_projection=data_projection, data_x_size=data_x_size, data_y_size=data_y_size)
+
+
+def _get_raster_data_source(key: str, s3_data, options: Optional[GeospatialOptions]):
+    """
+    Retrieves geospatial file from S3. If the options include Vector2RasterOptions, it will transform
+    the vector into a raster and return the raster datasource. If no Vector2RasterOptions are included,
+    it will assume the file is a raster and try to read it as a raster. The caller is expected to know whether
+    the geospatial file is a raster or vector and supply the appropriate options.
+
+    :param key: the object store key pointing to the desired geospatial file
+    :param s3_data: the object store content blob
```

Comment:
is s3_data the 'destination' for the file to be opened? Is it a variable or a file path?